src/Search/FieldToken.php

```diff
@@ -0,0 +1,67 @@
+<?php
+
+namespace App\Search;
+
+use App\Entity\User;
+
+class FieldToken implements TokenInterface
+{
+    private string $filterString = '';
+
+    #[\Override]
+    public function getRegex(): string
+    {
+        return '/[a-zA-Z_][a-zA-Z0-9_-]*:"[^"]+"/';
+    }
+
+    #[\Override]
+    public function setUser(User $user): void
+    {
+    }
+
+    #[\Override]
+    public function parseTokens(array $tokens): void
+    {
+        $filters = [];
+        foreach ($tokens as $token) {
+            [$field, $value] = explode(':', (string) $token);
+            $filters[$field][] = '`'.trim($value, '"').'`';
+        }
+
+        $filtered = [];
+        foreach ($filters as $field => $values) {
+            if ($field === 'orderBy') {
+                continue;
+            }
+            $trimmedfield = trim($field, '-');
+
+            $escapedValues = implode(',', $values);
+            $escapedValues = '['.$escapedValues.']';
```

Comment:
You could do a `json_encode((array)$values)` to avoid the `[` and ` ,` manual handling.